### ISSUE TITLE
Simplex2D can cause a world crash

### ISSUE THREAD

**Issue by BaxterOttoman** [Yellow-Dog-Man/Resonite-Issues#329]
### Describe the bug?

Simplex 2D can cause a world crash in several very normal situations.

### To Reproduce

Feed a 2D point generator into the input of a Simplex2D node, draw out the display, and the world crashes.

This works with other inputs too. I first discovered it by feeding a repacked raycast once position into it.

### Expected behavior

The node should return the output of the simplex function.

### Resonite Version Number

Beta 2023.10.18.449

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop

### Log Files

[PRODUCTION01 - 2023.10.18.449 - 2023-10-19 13_30_25.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13046510/PRODUCTION01.-.2023.10.18.449.-.2023-10-19.13_30_25.log)

### Reporters

@Baxter

**Comment by Frooxius** [comment]
Are you able to narrow down which input values cause the crash?

**Comment by BaxterOttoman** [comment]
Sorry I failed to see this earlier. I have used as input point on/in square and a float3 value from a raycast one repacked into a float2 and those both crashed the world. I can demonstrate for you if you'd like.

**Comment by Nytra** [comment]
You can replicate this if you provide an input of (X: -1, Y: 0) to the Simplex2D node and then pull out a display from it

**Comment by BaxterOttoman** [comment]
Yeah, as @Nytra says, you can just connect this and it causes a world crash.
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/15233062/4eac2b92-90b8-42f5-adb3-cd6680332451)

**Comment by 5H4D0W-X** [comment]
This has been fixed so this issue can be closed @shiftyscales

**Comment by Nytra** [comment]
@shiftyscales I believe it was fixed in Beta 2024.3.12.87

> Fixed crash from SimplexNoise2D when it's fed certain negative values (fixed by @ProbablePrime, reported by @Euphoria via security report ticket)
